src/test/java/com/example/javaee_jpa_hibernate/model/InvoiceTest.java

```diff
@@ -0,0 +1,169 @@
+package com.example.javaee_jpa_hibernate.model;
+
+import com.example.javaee_jpa_hibernate.model.counterparty.Address;
+import com.example.javaee_jpa_hibernate.model.counterparty.Counterparty;
+import com.example.javaee_jpa_hibernate.model.invoice_item.InvoiceItem;
+import com.example.javaee_jpa_hibernate.test_helper.InvoiceBuilder;
+import com.example.javaee_jpa_hibernate.test_helper.InvoiceGenerator;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class InvoiceTest {
+    @Test
+    public void shouldReturnDefaultInvoice() throws Exception {
+        //when
+        Invoice invoice = new Invoice();
+        //then
+        assertNotNull(invoice);
+    }
+
+    @Test
+    public void shouldReturnInvoice() throws Exception {
+        //given
+        LocalDate date = LocalDate.now();
+        Counterparty counterparty = InvoiceGenerator.generateCounterparty();
+        List<InvoiceItem> invoiceItems = InvoiceGenerator.generateInvoiceItems(1);
+        //when
+        Invoice invoice = InvoiceBuilder.builder()
+                .withDate(date)
+                .withCounterparty(counterparty)
+                .withInvoiceItems(invoiceItems)
+                .build();
+//        I did not know how to handle it:
+        invoice.setId(1L);
+        //then
+        assertNotNull(invoice);
+        assertEquals(1L, invoice.getId());
+        assertEquals(date, invoice.getDate());
+        assertEquals(counterparty, invoice.getCounterparty());
+        assertEquals(invoiceItems, invoice.getInvoiceItems());
+    }
+
+    @Test
+    public void shouldGetId() throws Exception {
```

Comment:
Could you please extend the test's scenario to include more details about what should this test verify?